crates/sui-replay/src/displays/transaction_displays.rs

```diff
@@ -184,8 +231,141 @@ impl<'a> Display for Pretty<'a, Argument> {
             Argument::GasCoin => "GasCoin".to_string(),
             Argument::Input(i) => format!("Input  {}", i),
             Argument::Result(i) => format!("Result {}", i),
-            Argument::NestedResult(j, k) => format!("Nested Result {}: {}", j, k),
+            Argument::NestedResult(j, k) => format!("Result {}: {}", j, k),
         };
         write!(f, "{}", output)
     }
 }
+impl<'a> Display for Pretty<'a, ResolvedResults> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let Pretty(ResolvedResults {
+            mutable_reference_outputs,
+            return_values,
+        }) = self;
+
+        let len_m_ref = mutable_reference_outputs.len();
+        let len_ret_vals = return_values.len();
+
+        if len_ret_vals > 0 {
+            write!(f, "\n Return Values:\n ──────────────")?;
+        }
+
+        for (i, value) in return_values.iter().enumerate() {
+            write!(f, "\n • Result {i:<2} ")?;
+            write!(f, "\n{}\n", value)?;
+        }
+
+        if len_m_ref > 0 {
+            write!(
+                f,
+                "\n Mutable Reference Outputs:\n ──────────────────────────"
+            )?;
+        }
+
+        for (arg, value) in mutable_reference_outputs {
+            write!(f, "\n • {} ", arg)?;
+            write!(f, "\n{}\n", value)?;
+        }
+
+        if len_ret_vals == 0 && len_m_ref == 0 {
+            write!(f, "\n No return values")?;
+        }
+
+        Ok(())
+    }
+}
+
+impl<'a> Display for Pretty<'a, TypeTag> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let Pretty(type_tag) = self;
+        match type_tag {
+            TypeTag::Vector(v) => {
+                write!(f, "Vector of {}", Pretty(&**v))
+            }
+            TypeTag::Struct(s) => {
+                write!(f, "{}::{}", s.module, s.name)
+            }
+            _ => {
+                write!(f, "{}", type_tag)
+            }
+        }
+    }
+}
+
+fn resolve_to_layout(
+    type_tag: &Box<TypeTag>,
+    executor: &Arc<dyn Executor + Send + Sync>,
+    store_factory: &LocalExec,
```

Comment:
Ooh! I tried that, along with several other approaches that involved saving the layout_resolver on a struct, or the ingredients to create the layout_resolver. Each time I tried that, I ran into an issue with lifetimes, ie for the executor and the store, I would see `borrowed value does not live long enough`,  `cast requires that executor is borrowed for 'static` etc. because the lifetime constraints on the `type_layout_resolver` function.
```
trait Executor {
    fn type_layout_resolver(
        &'static self,
        store: Box<dyn TypeLayoutStore + 'static>,
    ) -> Box<dyn LayoutResolver + 'static>;
}
```

It is actually called store_factory because at first it was a method on the `LocalExec` you could call to get a cloned instance of `Self` to use in the `resolve_value` and recursive `resolve_to_layout` functions, but passing in like this was the most simple way to use construct a resolver without storing these values anywhere!

Reply:
Thanks for the explanation! That's an unfortunate (but understandable) lifetime constraint on `type_layout_resolver`...